### ISSUE TITLE
current_shuffled_witnesses is a hex string in get_state

### ISSUE THREAD

**Issue by theoreticalbts** [steemit/steem#828]
Fix for #681 didn't `get_state`.  We need to properly peer the `witness_schedule_object` with a `witness_schedule_api_obj` that serializes properly instead of just shortcutting with a `typedef`.

**Comment by mvandeberg** [comment]
Related #657 #825

**Comment by theoreticalbts** [comment]
Has this been fixed, or is it still an issue?

**Comment by mvandeberg** [comment]
It looks like it is still an issue

**Comment by Kiwonik** [comment]
Question 1 - Where is witness_schedule_api_obj type defined? I can only see it used in condenser_api/state.hpp::121.

**Comment by Kiwonik** [comment]
Ad Q1 - Although we followed the instructions in _doc/building.md_ we don't have _app_ subdirectory of _steemit/steem_ referenced e.g. in `#include <steemit/app/steem_api_objects.hpp>`

**Comment by Kiwonik** [comment]
Question 2 - What are the steps to reproduce the error (any test created?) and what is the desired output (format)?

**Comment by mvandeberg** [comment]
The file in question is now `plugins/apis/database_api/include/ . . . /database_api_objects.hpp>`

Our database lives in a shared memory segments with its own memory allocator. To be efficient, we try to make as many fields as possible static sizes. We used to use a string for account names, now we used a fixed length string field that we can sort by integer comparison. That class is defined in `protocol/include/ . . . /fixed_string.hpp`. Because the base type is an integer, by default it serializes to an integer. There is a to_string method that will convert it back to a `std::string`, which we need to do whenever we return the field in an API.

This issue is to create an API version of the `witness_schedule_object`, called `api_witness_schedule_object`. Have a default constructor and a copy constructor taking the internal `witness_schedule_object`. Look at other classes in `database_api_objects.hpp` to see the pattern we are using. Most fields will be direct copies, however the witness schedule field needs to convert each of the `fixed_string` witness names to `std::string`.
